packages/database/src/sync/buildSyncLookupSelect.ts

```diff
@@ -9,18 +9,32 @@ interface Columns {
   isLabRequestValue?: string;
 }
 
-export function buildSyncLookupSelect(model: typeof Model, columns: Columns = {}) {
+/**
+ * When fully rebuilding we want to use the underlying
+ */
+const updatedAtSyncTickClause = (table: string, isFullyRebuilding: boolean) => {
+  const historicRecordSyncTick = `${table}.updated_at_sync_tick`;
+  const newRecordSyncTick = `COALESCE(:updatedAtSyncTick, ${table}.updated_at_sync_tick)`;
+
+  return isFullyRebuilding
+    ? `CASE WHEN ${table}.updated_at_sync_tick <= :since THEN ${historicRecordSyncTick} ELSE ${newRecordSyncTick} END`
+    : newRecordSyncTick;
```

Comment:
I think it might be worth putting a comment explaining this one

Reply:
Yeah that's a really good call, will do :+1: